services/motion/builtin/state/state.go

```diff
@@ -448,34 +438,38 @@ func (s *State) PlanHistory(req motion.PlanHistoryReq) ([]motion.PlanWithStatus,
 	// last plan only
 	if req.LastPlanOnly {
 		if ex := cs.lastExecution(); executionID == uuid.Nil || executionID == ex.id {
-			history := make([]motion.PlanWithStatus, 1)
-			copy(history, ex.history)
-			return history, nil
+			return renderableHistory(ex.history[:1]), nil
 		}
 
 		// if executionID is provided & doesn't match the last execution for the component
 		if ex, exists := cs.executionsByID[executionID]; exists {
-			history := make([]motion.PlanWithStatus, 1)
-			copy(history, ex.history)
-			return history, nil
+			return renderableHistory(ex.history[:1]), nil
 		}
 		return nil, resource.NewNotFoundError(req.ComponentName)
 	}
 
 	// specific execution id when lastPlanOnly is NOT enabled
 	if executionID != uuid.Nil {
 		if ex, exists := cs.executionsByID[executionID]; exists {
-			history := make([]motion.PlanWithStatus, len(ex.history))
-			copy(history, ex.history)
-			return history, nil
+			return renderableHistory(ex.history), nil
 		}
 		return nil, resource.NewNotFoundError(req.ComponentName)
 	}
 
 	ex := cs.lastExecution()
-	history := make([]motion.PlanWithStatus, len(cs.lastExecution().history))
-	copy(history, ex.history)
-	return history, nil
+	r := renderableHistory(ex.history)
+	s.logger.Debug(r)
```

Comment:
Do we want to keep this debug statement?

Reply:
nope good catch